lib/membrane/video_compositor/scene.ex

```diff
@@ -19,4 +24,37 @@ defmodule Membrane.VideoCompositor.Scene do
   def empty() do
     %Scene{video_configs: %{}}
   end
+
+  @spec pads(t()) :: MapSet.t()
+  def pads(%__MODULE__{video_configs: video_configs}) do
+    video_configs
+    |> MapSet.new(fn {pad, _video_config} -> pad end)
+  end
+
+  @spec validate(t(), MapSet.t()) :: :ok
+  def validate(scene = %__MODULE__{video_configs: video_configs}, input_pads) do
+    scene_pads = Scene.pads(scene)
+
+    unless MapSet.subset?(scene_pads, input_pads) do
+      raise "The scene must include references only to VideoCompositor input pads. \n" <>
+              "Scene: #{inspect(scene)} \n" <>
+              "Scene pads: #{inspect(MapSet.to_list(scene_pads))} \n" <>
+              "Input pads: #{inspect(MapSet.to_list(input_pads))}"
+    end
+
+    video_configs
+    |> Map.to_list()
+    |> Enum.map(fn {pad, video_config} ->
+      case video_config do
+        %VideoConfig{} ->
+          :ok
+
+        _else ->
+          raise "The scene must include only #{VideoConfig} structs. \n" <>
```

Comment:
Same as above, you can use multiline strings here.
And another thing - i believe `#{VideoConfig}` will be printed as "Elixir.VideoConfig". What you need here is probably `#{inspect(VideoConfig)}` (in case `VideoConfig` is aliased somewhere)